tests/struct.rs

```diff
@@ -0,0 +1,99 @@
+use serde::{Deserialize, Serialize};
+use serde_bin_prot::to_writer;
+use std::f64::INFINITY;
+
+mod common;
+use common::TestCase;
+
+#[derive(Debug)]
+enum StructTestCases {
+    TestA(A),
+    TestB(B),
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct A {
+    x: i64,
+    y: f64,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct B {
+    y: b_inner,
+    z: Option<i64>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct b_inner {
+    w: i64,
+    x: i64,
+}
```

Comment:
I don't think rust has anonymous structs, but let me know if there is some other way to do this

Reply:
You're right, it doesn't! Creating another struct for an inner one is fine. Though i'd name structs capitalized.

Reply:
updated!